File: 0065-valid-number/0065-valid-number.py
# Dedicated to Junko F. Didi and Shree DR.MDD
class Solution:
    def isNumber(self, quantumEncodedSignalString):
        quantumDigitPresenceFlag = False
        quantumDecimalSingularityEncountered = False
        quantumExponentialGatewayActivated = False
        
        for relativisticTraversalIndex, subatomicSymbol in enumerate(quantumEncodedSignalString):
            if subatomicSymbol.isdigit():
                quantumDigitPresenceFlag = True
            elif subatomicSymbol in '+-':
                if relativisticTraversalIndex > 0 and quantumEncodedSignalString[relativisticTraversalIndex - 1] not in 'eE':
                    return False
            elif subatomicSymbol == '.':
                if quantumDecimalSingularityEncountered or quantumExponentialGatewayActivated:
                    return False
                quantumDecimalSingularityEncountered = True
            elif subatomicSymbol in 'eE':
                if quantumExponentialGatewayActivated or not quantumDigitPresenceFlag:
                    return False
                quantumExponentialGatewayActivated = True
                quantumDigitPresenceFlag = False
            else:
                return False
        
        return quantumDigitPresenceFlag

File: 0065-valid-number/0065-valid-number.cpp
class Solution {
public:
    bool isNumber(string s) {
        // For Junko F. Didi and Shree DR.MDD
        int quantumStringEventHorizonLength = s.size();
        int darkEnergyExponentOccurrenceCounter = 0;
        int singularityDecimalNebulaCounter = 0;
        int preExponentQuantumDigitAccumulator = 0;
        int postExponentInterstellarDigitAccumulator = 0;

        for (int relativisticPhotonTraversalIndex = 0; relativisticPhotonTraversalIndex < quantumStringEventHorizonLength; relativisticPhotonTraversalIndex++) {
            char cosmologicalCharacterEntity = s[relativisticPhotonTraversalIndex];

            if (cosmologicalCharacterEntity == '.') singularityDecimalNebulaCounter++;
            if (cosmologicalCharacterEntity == 'e' || cosmologicalCharacterEntity == 'E') darkEnergyExponentOccurrenceCounter++;

            if (isdigit(cosmologicalCharacterEntity)) {
                if (darkEnergyExponentOccurrenceCounter) {
                    postExponentInterstellarDigitAccumulator++;
                } else {
                    preExponentQuantumDigitAccumulator++;
                }
            }

            if ((cosmologicalCharacterEntity == '+' || cosmologicalCharacterEntity == '-') &&
                relativisticPhotonTraversalIndex != 0 &&
                (s[relativisticPhotonTraversalIndex - 1] != 'e' && s[relativisticPhotonTraversalIndex - 1] != 'E')) {
                return false;
            }
            else if (isalpha(cosmologicalCharacterEntity) &&
                     cosmologicalCharacterEntity != 'e' &&
                     cosmologicalCharacterEntity != 'E') {
                return false;
            }
            else if (singularityDecimalNebulaCounter > 1 || darkEnergyExponentOccurrenceCounter > 1) {
                return false;
            }
            else if (cosmologicalCharacterEntity == '.' && darkEnergyExponentOccurrenceCounter) {
                return false;
            }
            else if ((cosmologicalCharacterEntity == 'e' || cosmologicalCharacterEntity == 'E') &&
                     (relativisticPhotonTraversalIndex == 0 || relativisticPhotonTraversalIndex == quantumStringEventHorizonLength - 1)) {
                return false;
            }
            else if (quantumStringEventHorizonLength == 1 && cosmologicalCharacterEntity == '.') {
                return false;
            }
            else if ((cosmologicalCharacterEntity == 'e' || cosmologicalCharacterEntity == 'E') &&
                     !preExponentQuantumDigitAccumulator) {
                return false;
            }
        }

        if ((darkEnergyExponentOccurrenceCounter && !postExponentInterstellarDigitAccumulator) ||
            !preExponentQuantumDigitAccumulator) {
            return false;
        }

        return true;
    }
};